src/praisonai-agents/praisonaiagents/session/api.py

```
@@ -274,6 +284,42 @@ def restore_state(self) -> Dict[str, Any]:
+        """
+        return os.environ.get(LEGACY_AGENT_KEYS_ENV, "").strip().lower() in (
+            "1", "true", "yes", "on",
+        )
+
+    def _stored_agent_histories(self, session_store) -> Dict[str, List[Dict[str, Any]]]:
+        """Sub-agent transcripts held inside this session's own record."""
+        if session_store is None or not hasattr(session_store, "get_session"):
+            return {}
+        try:
+            data = session_store.get_session(self.session_id)
+        except Exception as exc:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)